Continue with forward proxy if ReverseOnly is not true and no mapping available

sleemburg opened the pull request:
Hi,

As indicated by Michael Adam, a new pull request with rewritten code.

Kind regards,
Stephan

obnoxxx commented on src/reqs.c:
```diff
@@ -369,12 +369,15 @@ static struct request_s *process_request (struct conn_s *connptr,
+                if (reverse_url != NULL) {
+                        safefree (url);
+                        url = reverse_url;
+		} else if (config.reverseonly) {
```
There is a whitespace error here: TP currently uses spaces, not tabs.

obnoxxx commented on src/reqs.c:
```diff
@@ -369,12 +369,15 @@ static struct request_s *process_request (struct conn_s *connptr,
+                        url = reverse_url;
+		} else if (config.reverseonly) {
+                        indicate_http_error (connptr, 400, "Bad Request",
+                                     "detail", "No mapping found for requested url",
```
Formatting: please indent the arguments to match the opening bracket, and make sure the line length does not exceed 80 characters, if possible. i.e.

```
    func (connptr, ...
          "detail", "No ...."
          ...);
```

obnoxxx commented:
The patch looks good in general, but see the formatting comments in the patch.

also copying the comment from #32:



The diff looks good now. The two patches named "allow non-reverse mappings if reverseonly is not enabled" should be squashed. But I think for correctness the patch should modify reverse_rewrite_url() in src/reverse-proxy.c:160ff to remove the indicate_http_error call there and also the check for config.reverseonly. so that the hunk reads roughly like this:

```
   if (rewrite_url == NULL) {
        return NULL;
   }
```
because you put all the 'reverse_only' logic into the caller. am I correct?

sleemburg commented:
I made some changes, is it ok now?

obnoxxx commented on src/reverse-proxy.c:
```diff
@@ -166,7 +166,8 @@ char *reverse_rewrite_url (struct conn_s *connptr, hashmap_t hashofheaders,
         }
 
-        log_message (LOG_CONN, "Rewriting URL: %s -> %s", url, rewrite_url);
+        log_message (LOG_CONN, "Rewriting URL: %s -> %s", url, 
```
There is a trailing space at the end of the line, after ` url, `. That should be removed.

obnoxxx commented on src/reverse-proxy.c:
```diff
@@ -166,7 +166,8 @@ char *reverse_rewrite_url (struct conn_s *connptr, hashmap_t hashofheaders,
 
-        log_message (LOG_CONN, "Rewriting URL: %s -> %s", url, rewrite_url);
+        log_message (LOG_CONN, "Rewriting URL: %s -> %s", url, 
+			rewrite_url ? rewrite_url : "No mapping found");
```
The indentation is done with tabs here. Should be spaces to be consistent with the rest of the code.
And should be aligned with the opening bracket.

obnoxxx commented:
There are still formatting issues, and see the comment for the proposed change to reverse_rewrite_url() to remove the call to `indicate_http_error` from `reverse_rewrite_url`. I could do that for you, but want to give you the opportunity to do the changes yourself. This iterative review<->amend-patch cycle is quite common.

sleemburg commented:
I changed it, but forgot to do a push origin after the commit.. Still used to subversion..

I removed the indicate_http_error and the formatting with the tabs and null dereference is undone, as the null dereference is not possible anymore with the current flow.

Thank you for your help. I prefer to do the changes myself and learn from this review/amend-patch cycle.

obnoxxx commented on src/reqs.c:
```diff
@@ -372,10 +372,13 @@ static struct request_s *process_request (struct conn_s *connptr,
                         url = reverse_url;
-		} else if (config.reverseonly) {
+                } else if (config.reverseonly) {
+                        log_message (LOG_ERR, 
```
There is 1 trailing space at the end. Please remove.

obnoxxx commented on src/reqs.c:
```diff
@@ -372,10 +372,13 @@ static struct request_s *process_request (struct conn_s *connptr,
+                } else if (config.reverseonly) {
+                        log_message (LOG_ERR, 
+                                    "Bad request, no mapping for '%s' found",
+                                    url);
```
These need 1 space more of indentation: should be aligned not below the bracket but below the item in the bracket.

obnoxxx commented:
Thanks for updating! It's almost perfect now. Just two more whitespace errors.
The logic looks very clean and correct now.
Also good catch that the null-deref check is now not needed any more.

If you fix the whitespace errors, I'll be happy and merge.

obnoxxx commented on src/reqs.c:
```diff
@@ -373,7 +373,7 @@ static struct request_s *process_request (struct conn_s *connptr,
                         url = reverse_url;
                 } else if (config.reverseonly) {
-                        log_message (LOG_ERR, 
+                        log_message (LOG_ERR,
```
Great!

Now only indent the two following lines one more space and we're done. ;-)

sleemburg commented:
I fixed the whitespaces.

obnoxxx commented on src/reqs.c:
```diff
@@ -374,8 +374,8 @@ static struct request_s *process_request (struct conn_s *connptr,
-                                    "Bad request, no mapping for '%s' found",
-                                    url);
+                                     "Bad request, no mapping for '%s' found",
+                                     url);
```
perfect.

obnoxxx commented:
Merged.
Thanks a lot for your contribution!
Tinyproxy has become a bit better! :-)

sleemburg commented:
Thanks! Also for your help.

On 10-09-16 19:24, Michael Adam wrote:
>
> Merged.
> Thanks a lot for your contribution!
> Tinyproxy has become a bit better! :-)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tinyproxy/tinyproxy/pull/35#issuecomment-246124099>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQFkdm701lLMhnDkTW344MErHu2ks_Szks5qoue5gaJpZM4J5wnX>.
>
